katran/lib/PcapWriter.cpp

```diff
@@ -81,14 +81,14 @@ bool PcapWriter::writePcapHeader(uint32_t writerId) {
     LOG(ERROR) << "no writer w/ specified ID: " << writerId;
     return false;
   }
-  if (!dataWriters_[writerId]->available(sizeof(struct pcap_hdr_s))) {
-    LOG(ERROR) << "DataWriter failed to write a header. Too few space.";
-    return false;
-  }
   if (headerExists_[writerId]) {
     VLOG(4) << "header already exists";
     return true;
   }
+  if (!dataWriters_[writerId]->available(sizeof(struct pcap_hdr_s))) {
```

Comment:
One last thing - why don't we check `available(..)` before`dataWriters_[writerId]->writeData` in in line # 75?

Reply:
line 117. we check that we have enough available data before calling writePacket